R/theme_cmap.R

```diff
@@ -1,166 +1,191 @@
+      legend.text.align = 0,
+      legend.background = ggplot2::element_blank(),
+      legend.text = ggplot2::element_text(),
+      legend.title = ggplot2::element_blank(),
+      legend.key = ggplot2::element_blank(),
+
+      # Axis format
+      axis.title.y = ggplot2::element_blank(),
+      axis.title.x = ggplot2::element_blank(),
+      axis.text = ggplot2::element_text(color = cmapplot_globals$colors$blackish),
+      axis.text.x = ggplot2::element_text(margin = ggplot2::margin(5, b = 10)),
+      axis.ticks = ggplot2::element_blank(),
+      axis.line = ggplot2::element_blank(),
+
+      # Blank background
+      panel.background = ggplot2::element_blank(),
+      plot.background = ggplot2::element_blank(),
+
+      # No gridlines
+      panel.grid.major.x = ggplot2::element_blank(),
+      panel.grid.minor.x = ggplot2::element_blank(),
+      panel.grid.major.y = ggplot2::element_blank(),
+      panel.grid.minor.y = ggplot2::element_blank(),
+
+      # Strip background
+      strip.background = ggplot2::element_rect(fill = "white"),
+
+      # Facet wrap text
+      strip.text = ggplot2::element_text(hjust = 0)
+    ),
+
+    # The following elements get added based on the presence of specific
+    # function arguments. These elements add to or overwrite portions of
+    # the default theme.
+
+    # Re-introduce x label, if specified
+    if(!is.null(xlab)){
+      ggplot2::theme(axis.title.x = element_text())
+    },
+    if(!is.null(xlab)){
+      ggplot2::xlab(xlab)
+    },
+
+    # Re-introduce y label, if specified
+    if(!is.null(ylab)){
+      ggplot2::theme(axis.title.y = element_text())
+    },
+    if(!is.null(ylab)){
+      ggplot2::ylab(ylab)
+    },
+
+    # Add x origin line, if specified
+    if(!is.null(hline)){
+      ggplot2::geom_hline(yintercept = hline,
+                          size = ggplot_size_conversion(cmapplot_globals$plot_constants$lwd_originline),
+                          color = cmapplot_globals$colors$blackish)
+    },
+
+    # Add y origin line, if specified
+    if(!is.null(vline)){
+      ggplot2::geom_vline(xintercept = vline,
+                          size = ggplot_size_conversion(cmapplot_globals$plot_constants$lwd_originline),
+                          color = cmapplot_globals$colors$blackish)
+    },
+
+    # Re-introduce horizontal gridlines if specified
+    if (grepl("h", gridlines)) {
+      ggplot2::theme(
+        panel.grid.major.y = ggplot2::element_line(
+          size = ggplot_size_conversion(cmapplot_globals$plot_constants$lwd_gridline),
+          color = cmapplot_globals$colors$blackish)
+      )
+    },
+
+    # Re-introduce vertical gridlines if specified
+    if (grepl("v", gridlines)) {
+      ggplot2::theme(
+        panel.grid.major.x = ggplot2::element_line(
+          size = ggplot_size_conversion(cmapplot_globals$plot_constants$lwd_gridline),
+          color = cmapplot_globals$colors$blackish)
+      )
+    },
+
+    # FLAG FOR REVIEW - experimental functionality
```

Comment:
I didn't test this extensively, but it seemed to work fine when I did use it.

Anything else to do regarding this (other than cleaning up comments) before merging?

Reply:
No, I think it's fine. I wanted to make sure that the inclusion of both "col" and "fill" wasn't causing any issues on anyone else's machine, but it seems like it is working fine.

Reply:
@dlcomeaux was there something you were looking for here in particular?

`max_columns` as a name is not explicitly about the legend. It may be worth changing the name of the argument. I have no other thoughts on this, but I have not extensively tested it either.

Reply:
@matthewstern see my response to Noel above, I think it overlapped.

Open to changing the name. I'd say we deal with that once the whole thing is incorporated, though?

Reply:
Works for me!